### PR TITLE
fix unfiltered comparison in controller test

### LINKED ISSUE #34580: TestDeploymentController_dontSyncDeploymentsWithEmptyPodSelector {deployment}
https://k8s-gubernator.appspot.com/build/kubernetes-jenkins/logs/kubernetes-test-go/17996/

Failed: TestDeploymentController_dontSyncDeploymentsWithEmptyPodSelector {deployment}

```
deployment_controller_test.go:270: expected deployment controller to not take action
```

Previous issues for this test: #34547

### PR DESCRIPTION
**Pull request by deads2k**
fixes https://github.com/kubernetes/kubernetes/issues/34580

@piosz @kargakis fyi

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/34608)

### CODE REVIEW

File: pkg/controller/deployment/deployment_controller_test.go [hunk 1]
```diff
@@ -261,10 +260,12 @@ func TestDeploymentController_dontSyncDeploymentsWithEmptyPodSelector(t *testing
 	// We expect the deployment controller to not take action here since it's configuration
 	// is invalid, even though no replicasets exist that match it's selector.
 	controller.syncDeployment(fmt.Sprintf("%s/%s", d.ObjectMeta.Namespace, d.ObjectMeta.Name))
-	if len(fake.Actions()) == 0 {
```
Comment (deads2k) [inline pkg/controller/deployment/deployment_controller_test.go]: I missed this one before :(

**Comment by piosz** [comment]
@deads2k thanks for the PR. Unfortunately I don't understand the code enough to review it.

**Comment by kargakis** [comment]
LGTM

**Comment by piosz** [comment]
This is low risk change, so I'm merging manually to fix suite with unit tests.

**Pull request merged**